pandas/tests/groupby/test_groupby.py

```diff
@@ -3843,6 +3843,128 @@ def test_cummin_cummax(self):
+        df = pd.DataFrame(source_dict)
+        result = df.groupby(['B']).C.is_monotonic_increasing()
+        expected = pd.Series(index=['a', 'b', 'c', 'd'],
+                             data=[True, True, False, True],
+                             name='C')
+        expected.index.name = 'B'
+        expected.index.name = 'B'
+        tm.assert_series_equal(result, expected)
+        # Also check result equal to manually taking x.is_monotonic_increasing.
+        expected = df.groupby('B').C.apply(lambda x: x.is_monotonic_increasing)
+        tm.assert_series_equal(result, expected)
+
+        # As above, for .is_monotonic_decreasing()
+        # Basics: strictly decreasing (T), strictly increasing (F),
+        # abs val decreasing (F), non-strictly increasing (T)
+        source_dict = {
+            'A': ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11'],
+            'B': ['a', 'a', 'a', 'b', 'b', 'b', 'c', 'c', 'c', 'd', 'd'],
+            'C': [10, 9, 7, 3, 4, 5, -3, 2, 0, 1, 1]}
+        df = pd.DataFrame(source_dict)
+        result = df.groupby(['B']).C.is_monotonic_decreasing()
+        expected = pd.Series(index=['a', 'b', 'c', 'd'],
+                             data=[True, False, False, True],
+                             name='C')
+        expected.index.name = 'B'
+        tm.assert_series_equal(result, expected)
+        # Also check result equal to manually taking x.is_monotonic_decreasing.
+        expected = df.groupby('B').C.apply(lambda x: x.is_monotonic_decreasing)
+        tm.assert_series_equal(result, expected)
+
+        # Test with inf vals
+        source_dict = {
+            'A': ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11'],
+            'B': ['a', 'a', 'a', 'b', 'b', 'b', 'c', 'c', 'c', 'd', 'd'],
+            'C': [np.inf, 1, -np.inf, np.inf, 2, -3, -np.inf, 5, -3, -np.inf,
+                  -np.inf]}
+        df = pd.DataFrame(source_dict)
+        result = df.groupby(['B']).C.is_monotonic_decreasing()
+        expected = pd.Series(index=['a', 'b', 'c', 'd'],
+                             data=[True, True, False, True],
+                             name='C')
+        expected.index.name = 'B'
+        tm.assert_series_equal(result, expected)
+        # Also check result equal to manually taking x.is_monotonic_decreasing.
+        expected = df.groupby('B').C.apply(lambda x: x.is_monotonic_decreasing)
+        tm.assert_series_equal(result, expected)
+
+        # Test with nan vals; should always be False
+        source_dict = {
+            'A': ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11'],
+            'B': ['a', 'a', 'a', 'b', 'b', 'b', 'c', 'c', 'c', 'd', 'd'],
+            'C': [1, 2, np.nan, 3, 2, np.nan, np.nan, 5, -np.inf, 1, np.nan]}
+        df = pd.DataFrame(source_dict)
+        result = df.groupby(['B']).C.is_monotonic_decreasing()
+        expected = pd.Series(index=['a', 'b', 'c', 'd'],
+                             data=[False, False, False, False],
```

Comment:
FYI some of these may be less verbose as pytest parameterized tests, though not a big impact

Thanks for the PR!

Reply:
Thanks for the feedback. I did consider parametrizing, but when I've done it before, it inflated the test count (i.e. if I parametrized using the same test cases for both is_monotonic_increasing() and is_monotonic_decreasing, I'd have added two tests, rather than one. Maybe there's a way around that.)

Reply:
actually we would *prefer* to parametrize, it reduces the amount of duplication in code. feel free.